src/core/sys/posix/stdio.d

```diff
@@ -344,3 +344,7 @@ unittest
     assert(memcmp(ptr, testdata.ptr, testdata.length*wchar_t.sizeof) == 0);
     assert(fclose(f) == 0);
 }
+
+
+ssize_t getdelim (char** lineptr, size_t* n, int delimiter, FILE* stream);
+ssize_t getline (char** lineptr, size_t* n, FILE* stream);
```

Comment:
Using DMD 2.073.0 and compiling with `-dip1000` I got the following results:
```
void main() @safe
{
    char[100] buf;
    char* c;
    int n;

    bar(&c, &n); // Error
}

void bar(char** c, int* n) @safe
{
}
```

> dip1000.d(7): Error: reference to local variable c assigned to non-scope parameter c calling dip1000.bar
dip1000.d(7): Error: reference to local variable n assigned to non-scope parameter n calling dip1000.bar

```
void main() @safe
{
    char[100] buf;
    char* c;
    int n;

    bar(&c, &n); // OK
}

void bar(scope char** c, scope int* n) @safe
{
}
```

> [No errors]

```
void main() @safe
{
    char[100] buf;
    char* c = &buf[0];
    int n;

    bar(&c, &n);
}

void bar(scope char** c, scope int* n) @safe
{
}
```

> dip1000.d(7): Error: cannot take address of scope local c in @safe function main

(same result as yours)

So while DIP1000 is not a 100% solution (yet), using `scope` is still an improvement for `@safe` code, because it allows to pass pointers to *some* local variables.